src/main/com/yetanalytics/datasim/math/random.clj

```diff
@@ -1,211 +1,238 @@
 (ns com.yetanalytics.datasim.math.random
-  "Seeded random functions"
-  (:require [com.yetanalytics.datasim.util.maths :as maths]
-            [clojure.spec.alpha :as s]
-            [clojure.spec.gen.alpha :as sgen])
-  (:import [java.util
-            UUID Random]))
+  "Random number generation and probabilistic operations."
+  (:require [clojure.spec.alpha :as s]
+            [clojure.spec.gen.alpha :as sgen]
+            [com.yetanalytics.datasim.util.maths :as maths])
+  (:refer-clojure :exclude [rand rand-int rand-nth random-sample shuffle])
+  (:import [java.util UUID Random]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Specs
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
 (s/def ::seed int?)
 
 (s/def ::rng
   (s/with-gen #(instance? Random %)
-    (fn []
-      (sgen/fmap (fn [s]
-                   (Random. s))
-                 (s/gen ::seed)))))
+    (fn [] (sgen/fmap (fn [s] (Random. s)) (s/gen ::seed)))))
+
+(s/def ::sd
+  (s/double-in :min 0.0 :infinite? false :NaN? false))
+
+(s/def ::prob
+  (s/double-in :min 0.0 :max 1.0))
+
+(s/def ::weight
+  (s/double-in :min -1.0 :max 1.0))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Functions
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;; RNG Creation ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(s/fdef seed-rng
+  :args (s/cat)
+  :ret ::rng)
+
+(defn rng
+  "Create a pseudorandom RNG using an arbitrary seed value."
+  ^Random []
+  (Random.))
 
 (s/fdef seed-rng
   :args (s/cat :seed ::seed)
   :ret ::rng)
 
 (defn seed-rng
+  "Create a seeded deterministic, pseudorandom RNG, in which two RNGs created
+   using the same value of `seed` will output the same results."
   ^Random [^Long seed]
   (Random. seed))
 
-(s/fdef rand*
+;; Random Number Generation ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(s/fdef rand
   :args (s/cat :rng ::rng
                :n (s/? number?))
   :ret double?)
 
-(defn rand*
+(defn rand
```

Comment:
`rand*`, `rand-int*` etc were so named so they wouldn't collide with function names in core. Don't you get dreaded compilation warnings when they are named this way?

Reply:
`:refer-clojure :exclude` was my friend